### DISCUSSION TITLE
compile doris error

### DISCUSSION THREAD

**Discussion by superdiaodiao** [apache/doris#25343] [A - General / Q&A] [answered]
I followed the guild of this website, https://doris.apache.org/docs/dev/install/source-install/compilation-general/,
I used the latest image by
` docker pull apache/doris:build-env-ldb-toolchain-latest`
and compiled the lastest master branch of Doris
, then I got this,
`[ERROR] Plugin net.sourceforge.czt.dev:cup-maven-plugin:1.6-cdh or one of its dependencies could not be resolved: Could not find artifact net.sourceforge.czt.dev:cup-maven-plugin:jar:1.6-cdh in central (https://repo.maven.apache.org/maven2) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException`

Can someone give me a hand?

**Comment by yagagagaga** [comment]
This looks like some dependencies haven't been fully downloaded. You can check if it exists in this directory `~/.m2/repository/net/sourceforge/czt/dev/cup-maven-plugin/1.6-cdh`.

**Reply by superdiaodiao** [reply]
In the docker container, I can see these,
```
$ls ~/.m2/repository/net/sourceforge/czt/dev/cup-maven-plugin/1.6-cdh
cup-maven-plugin-1.6-cdh.jar.lastUpdated  cup-maven-plugin-1.6-cdh.pom.lastUpdated
```

**Reply by yagagagaga** [reply]
Normally, you should see the following content:
```sh
ls
_remote.repositories                     cup-maven-plugin-1.6-cdh.pom
cup-maven-plugin-1.6-cdh.jar             cup-maven-plugin-1.6-cdh.pom.lastUpdated
cup-maven-plugin-1.6-cdh.jar.lastUpdated cup-maven-plugin-1.6-cdh.pom.sha1
cup-maven-plugin-1.6-cdh.jar.sha1
```
If there are missing files, you can delete this directory and recompile it, so Maven will download these files again.

**Reply by superdiaodiao** [reply]
Could you give me a guild to recompile it? 

Or only I have to wait for the Doris to fix this docker problem?

**Reply by yagagagaga** [reply]
Yes, just run `bash build.sh --fe`, Doris will fix it.

**Reply by superdiaodiao** [reply]
I tried the command
`bash build.sh --fe`
and still got this,
`[ERROR] Plugin net.sourceforge.czt.dev:cup-maven-plugin:1.6-cdh or one of its dependencies could not be resolved: Failure to find net.sourceforge.czt.dev:cup-maven-plugin:jar:1.6-cdh in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException`

**Reply by yagagagaga** [reply]
Have you deleted `~/.m2/repository/net/sourceforge/czt/dev/cup-maven-plugin/1.6-cdh`?

**Reply by superdiaodiao** [reply]
yes, I ran this 
`rm -rf ~/.m2/repository/net/sourceforge/czt/dev/cup-maven-plugin/1.6-cdh`
and then ran `bash build.sh --fe`, but the error is the same.

**Reply by yagagagaga** [reply]
Your network seems unable to reach this package. You can manually place this package in that directory.
[1.6-cdh.zip](https://github.com/apache/doris/files/12878304/1.6-cdh.zip)

**Reply by superdiaodiao** [reply]
Thanks for the zip, but after I unzipped it, and my dir like this,
`ls ~/.m2/repository/net/sourceforge/czt/dev/cup-maven-plugin/1.6-cdh
_remote.repositories          cup-maven-plugin-1.6-cdh.jar.lastUpdated  cup-maven-plugin-1.6-cdh.pom              cup-maven-plugin-1.6-cdh.pom.sha1
cup-maven-plugin-1.6-cdh.jar  cup-maven-plugin-1.6-cdh.jar.sha1         cup-maven-plugin-1.6-cdh.pom.lastUpdated`
then I ran 
`sh build.sh --fe`, 
the error show up again.

**Reply by yagagagaga** [reply]
It shouldn't be happened! Try more again?

**Reply by superdiaodiao** [reply]
still not working, very strange.
I tried to use a new container but failed either.
Can you reproduce the error by using the latest image and code?

**Reply by yagagagaga** [reply, marked as answer]
1. pull latest docker image
```sh
docker run -it \
        --privileged=true \
        -v ~/.m2:/root/.m2 \
        -v ~/.npm:/root/.npm \
        --name doris-build-env \
        apache/doris:build-env-ldb-toolchain-latest
```
2. enter docker container and pull doris source code
```sh
git pull git@github.com:apache/doris.git
```
3. download prebuild thirdparty
```sh
wget https://github.com/apache/doris-thirdparty/releases/download/automation/doris-thirdparty-prebuilt-linux-x86_64.tar.xz
tar xfJ doris-thirdparty-prebuilt-linux-x86_64.tar.xz
mv install doris/thirdparty
```
4. build doris
```sh
cd doris
bash build.sh 
```

I can compile successfully.
<img width="1014" alt="image" src="https://github.com/apache/doris/assets/21151126/3441b50d-cc04-47b1-be7a-8ff2972c3ed9">

**Reply by superdiaodiao** [reply]
Thank you.
Is that the 3rd step neccessary?

Also, when I ran `mvn install doris/thirdparty`
the error is, 
`[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/root). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException`

**Reply by yagagagaga** [reply]
Yes, thirdparty library is required. And `mvn install doris/thirdparty` is an incorrect command

**Reply by superdiaodiao** [reply]
Problem solved.
It turns out it was my maven setting problem. I fixed it.
Thank you very much.
